barretenberg/cpp/src/barretenberg/vm2/simulation/lib/sha256_compression.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <cstdint>
+#include <span>
+
+#include "barretenberg/vm2/common/field.hpp"
+
+namespace bb::avm2::simulation {
+
+std::array<uint32_t, 8> sha256_block(const std::array<uint32_t, 8>& h_init, const std::array<uint32_t, 16>& input);
+constexpr uint32_t ror(uint32_t val, uint32_t shift);
```

Comment:
For this to actually be `constexpr` (or inlined) you probably have to move it from the cpp to here. Otherwise you might actually be dispatching calls.